### PR TITLE
Improve documentation on methods.py and integrate.py

### PR DESCRIPTION
**Pull request by syjlee**
## Description
Adding and Editing the documentation on methods.py and integrate.py.
The documentation covers public methods and properties of classes.
The private methods are commented with `# noqa` to mark as undocumented.
Mainly args and attributes are taken care of to improve.
Currently working on rephrasing descriptions on args and attributes and also on extended params on Langevin and Brownian.

## Motivation and Context
To resolve the Issue #721 .

## How Has This Been Tested?
Tested locally generated doc html through sphinx

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).
- [ ] My name is on the [list of contributors](https://github.com/glotzerlab/hoomd-blue/blob/master/sphinx-doc/credits.rst).

### CODE REVIEW

**Comment by syjlee** [comment]
I couldn't figure out the two warnings from Sphinx running since the number next to the file is not absolute line number. What would be the good way to debug if I'm not aware of all the docstring rules?
```Running Sphinx v3.2.1
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 0 source files that are out of date
updating environment: 0 added, 1 changed, 0 removed
reading sources... [100%] module-md-methods
/Users/syjlee/code/hoomd-blue/hoomd/md/methods.py:docstring of hoomd.md.methods.Brownian:142: WARNING: Inline literal start-string without end-string.
/Users/syjlee/code/hoomd-blue/hoomd/md/methods.py:docstring of hoomd.md.methods.Langevin:143: WARNING: Inline literal start-string without end-string.
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] package-md
generating indices...  genindex py-modindexdone
writing additional pages...  searchdone
copying static files... ... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build finished with problems, 2 warnings.```

**Comment by vyasr** [comment]
@syjlee yes, those errors are unfortunately not very helpful. The first problem is that line numbers are relative to the start of the docstring, not the file. The second problem is that the errors like this often only get reported at the very end of the docstring because of the way Sphinx works, for example with Brownian if you look the docstring probably has exactly 142 lines and the error is just being reported on the final line.

From a quick look, my guess would be that the error comes from line 854, where you have a citation in backticks that is split over multiple lines. Try combining all of those into one line and see if it resolves the problem. If it does, you'll get a separate error about line lengths being too long, which you could fix by adding a comment `#noqa: E501` to the end of the docstring (unless @joaander or @b-butler have a preferred solution).

To your more general question of how to find these errors in docstrings, I have honestly never found a great solution to this. I tend to just do a manual binary search where I delete a bunch of lines and see if the error goes away, then narrow it down to the problematic line... but perhaps the others have a more efficient solution they can provide.

**Comment by syjlee** [comment]
@vyasr Thanks to your comment, I could narrow down 'line number' by creating errors at multiple places in the same class and found out the error came from an unintended space between ending double tick and parameter name. Now I'm not seeing any warning or error messages with Sphinx run, but I could see some problems from actual generated html page.
For the attributes `gamma` and `gamma_r` in Brownian and Langevin, The type definition of *float or tuple of float* inside of TypeParameter doesn't have link unlike other attributes. I'm not quite sure what I missed in the format.
Also, now the attach function in feature/new-object-API is private method, so I need to resolve the conflict. What git command would be able to fix that?

**Comment by syjlee** [comment]
I found out that space should be entered between float element in the tuple in the typeParameter.
So far, I'm not finding formatting problem from html.
Let me know things should be revised not only formatting but also contents and descriptions.
Just for the quick questions, I'm still seeing doc page of methods.py and integrate.py are not seen with their name. there module name is still same as version 2. I wonder where in file controls the index name of methods.py and integrate.py

File: hoomd/md/integrate.py
Comment (b-butler) [inline hoomd/md/integrate.py:21]: We should start the function name with an underscore.

File: hoomd/md/integrate.py
Comment (b-butler) [inline hoomd/md/integrate.py:30]: We should start the function name with an underscore.

File: hoomd/md/integrate.py
Comment (b-butler) [inline hoomd/md/integrate.py:92]: Make mention that the intersection of the subsets must be null and the union must come to all the particles.

File: hoomd/md/integrate.py
Comment (b-butler) [inline hoomd/md/integrate.py:99]: Define what is meant by auto detect.

File: hoomd/md/integrate.py
Comment (b-butler) [inline hoomd/md/integrate.py:113]: Remove these until #666 is merged.

File: hoomd/md/integrate.py
Comment (b-butler) [inline hoomd/md/integrate.py:127]: Remove dem for now.

File: hoomd/md/integrate.py
Comment (b-butler) [inline hoomd/md/integrate.py:135]: Show the creation of an nve and lj object.

File: hoomd/md/integrate.py
Comment (b-butler) [inline hoomd/md/integrate.py:156]: Keep `None` as the default for forces, constraints, and methods. Method defaults need to be immutable in Python otherwise they can be changed which is not desired.

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:59]:
```suggestion
        nvt=hoomd.md.methods.NVT(filter=hoomd.filter.All(), kT=1.0, tau=0.5)
```

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:718]: This uses the Brownian method.

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:721]: These are type parameters so to set for all types change the default (i.e. `langevin.gamma.default = 2.0`).

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:895]: See comment on Langevin above

File: sphinx-doc/module-md-constrain.rst
Comment (b-butler) [inline sphinx-doc/module-md-constrain.rst:25]: These have not be converted so do not document them.

File: sphinx-doc/module-md-force.rst
Comment (b-butler) [inline sphinx-doc/module-md-force.rst:23]: Only document `Active` out of these.

File: sphinx-doc/module-md-methods.rst
Comment (b-butler) [inline sphinx-doc/module-md-methods.rst:32]: Only document classes that have been converted over.

**Review by b-butler — CHANGES_REQUESTED** [review]
Thanks for the work @syjlee. I have left some comments for you to address.

File: hoomd/md/integrate.py
Comment (joaander) [reply hoomd/md/integrate.py:92]: The union does not need to be all particles.

**Comment by joaander** [comment]
Merge `feature/new-object-API` into this branch and you can document `NPT`. Sphinx will also no longer warn about invalid references, so removing unimplemented features will not result in invalid reference warnings scattered about the code base.

**Review by joaander — CHANGES_REQUESTED** [review]
Great! We need a few small changes to use the renamed functions so that the tests pass again and a few minor edits to the docs.

**Comment by syjlee** [comment]
- Minor changes were made based on the review.
@joaander 
Please leave comments if you find more places to be fixed.

**Review by joaander — APPROVED** [review]
Thanks! This looks great!

File: hoomd/md/integrate.py
Comment (b-butler) [inline hoomd/md/integrate.py:127]: Remove until ported.

File: hoomd/md/integrate.py
Comment (b-butler) [inline hoomd/md/integrate.py:157]: I would like to see the same documentation for the attributes as for the arguments here.

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:56]: Does `#TODO` work here? rst comments begin with `..`

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:120]:
```suggestion
        S (`list`[`hoomd.variant.Variant`] or `float`): Stress components set 
```

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:130]:
```suggestion
        box_dof(`list`[`bool`]): Box degrees of freedom with six boolean elements 
```

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:182]:
```suggestion
    ``[True,True,True,False,False,False]``
```

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:204]: If it is unpublished and not to be published should we remove this?

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:243]: Mention being able to set an isotropic pressure using `npt.S = 4.`. or any arbitrary float.

File: hoomd/md/integrate.py
Comment (syjlee) [reply hoomd/md/integrate.py:127]: @b-butler I can remove `nph`, but isn't NPT already ported?

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:587]: We aren't using the C++ logging system, so this actually does nothing for now, because we don't expose the quantity to be logged in Python. @joaander would you rather wait and just remove this from the constructor or expose the reservoir energy to be logged?

**Review by b-butler — CHANGES_REQUESTED** [review]
Thanks @syjlee I left some comments for you to address.

File: hoomd/md/integrate.py
Comment (b-butler) [reply hoomd/md/integrate.py:127]: Yes, but we didn't port `NPH` in Python.

File: hoomd/md/methods.py
Comment (joaander) [reply hoomd/md/methods.py:587]: We can remove it from here in this PR. I'll add the loggable quantity back in on another PR and it will get documented there.

File: hoomd/md/methods.py
Comment (joaander) [reply hoomd/md/methods.py:204]: @jglaser How would you like us to document this going forward?

File: hoomd/md/integrate.py
Comment (syjlee) [reply hoomd/md/integrate.py:157]: @b-butler Same documentation means all the data type or descriptions as arguments? Descriptions on arguments have extra information of default value, comparing to Attributes.

Comment (joaander) [reply hoomd/md/integrate.py:157]: The sections should be mostly the same. Some the the exceptions are attributes that are not constructor arguments, type information where attributes have explicit types and arguments may be duck typed, and as you mention defaults.

File: hoomd/md/methods.py
Comment (syjlee) [reply hoomd/md/methods.py:56]: @b-butler  I thought this is general way how dev people can get aware of future tasks. Are we keeping official documentation for todos too?
With legit rst formatting`.. todo:: `, generated html doesn't show the content, which seems good to me.

File: hoomd/md/methods.py
Comment (syjlee) [reply hoomd/md/methods.py:587]: I removed it.

File: hoomd/md/integrate.py
Comment (syjlee) [reply hoomd/md/integrate.py:156]: @joaander right, do you think I need to fix the data types on the attributes? `methods`, `forces`, `constraints` are returning their values with `list` type. and `dt` and `aniso` are returned into `float` and `str` type each.

File: sphinx-doc/module-md-force.rst
Comment (joaander) [inline sphinx-doc/module-md-force.rst:11]: Please resolve these merge conflicts. Accept the changes from `feature/new-object-API`.

**Comment by DomFijan** [comment]
We had a short discussion this week on developers meeting that it would be helpful if documentation for nvt and npt integrators contained the information about suggested values for coupling parameters (tau and tauP for nvt and npt) similar to how it is done in LAMMPS documentation (See examples below). They recommend 100 time steps for tau and 1000 time steps for tauP for "many models".
image
image

**Comment by syjlee** [comment]
@DomFijan Thanks for the suggestions, and I added the note section for recommended value of coupling constants `tau` and `tauS` (Tdamp and Pdamp in case of LAMMPS). I'm waiting for further review including this to check if there's any misinformation or more to add.

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:60]: I would prefer just recommending `100 * dt` rather than the notation of 100 time steps.

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:128]:
```suggestion
        S (`list`[`hoomd.variant.Variant`] or `float`): Stress components set 
```

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:138]:
```suggestion
        box_dof(`list`['bool']): Box degrees of freedom with six boolean elements 
```

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:231]: See previous comment on thermostat.

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:259]:
```suggestion
            In Voigt notation: :math:`[S_{xx}, S_{yy}, S_{zz}, S_{yz}, S_{xz}, S_{xy}]`. Stress can be 
```

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:262]:
```suggestion
            pressure can be set by ``npt.S = 4.``
```

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:128]:
```suggestion
        S (`list`[`hoomd.variant.Variant`] or `float`): Stress components set 
```

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:132]:
```suggestion
            In case of isotropic pressure P (:math:`[p, p, p, 0, 0, 0]`), use ``S = p``.
```

File: hoomd/md/methods.py
Comment (b-butler) [inline hoomd/md/methods.py:139]:
```suggestion
        box_dof(`list`[`bool`]): Box degrees of freedom with six boolean 
```

**Review by b-butler — CHANGES_REQUESTED** [review]
I have only a few comments to address. Since they are suggestions, I will apply them and approve the PR.

**Review by b-butler — APPROVED** [review]

**Comment by joaander** [comment]
Thanks!

**Pull request merged**
